apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -191,44 +190,52 @@ export class MessageMatcher {
       payload.subscriber = variables.subscriber;
     } else {
       payload.subscriber = await this.subscriberRepository.findOne({
-        _id: configuration.command.subscriberId,
-        _environmentId: configuration.command.environmentId,
+        _id: command.subscriberId,
+        _environmentId: command.environmentId,
       });
     }
 
     if (variables.payload) {
       payload.payload = variables.payload;
     }
 
-    payload.identifier = configuration.command.identifier;
-    payload.channel = configuration.command.job.type;
+    payload.identifier = command.identifier;
+    payload.channel = command.job.type;
 
-    if (configuration.command.job.providerId) {
-      payload.providerId = configuration.command.job.providerId;
+    if (command.job.providerId) {
+      payload.providerId = command.job.providerId;
     }
 
     return payload;
   }
 
-  private async buildHmac(configuration: IMessageFilterConfiguration): Promise<string> {
-    if (process.env.NODE_ENV === 'test' && !configuration) return '';
+  private async buildHmac(command: SendMessageCommand): Promise<string> {
+    if (process.env.NODE_ENV === 'test') return '';
 
     const environment = await this.environmentRepository.findOne({
-      _id: configuration.command.environmentId,
-      _organizationId: configuration.command.organizationId,
+      _id: command.environmentId,
+      _organizationId: command.organizationId,
     });
 
-    return createHmac('sha256', environment.apiKeys[0].key).update(configuration.command.environmentId).digest('hex');
+    return createHmac('sha256', environment.apiKeys[0].key).update(command.environmentId).digest('hex');
   }
 
-  private async processFilter(variables: IFilterVariables, child, configuration: IMessageFilterConfiguration) {
+  private async processFilter(
+    variables: IFilterVariables,
+    child: FilterParts,
+    command: SendMessageCommand
+  ): Promise<boolean> {
     if (child.on === 'webhook') {
-      const res = await this.getWebhookResponse(child, variables, configuration);
+      const res = await this.getWebhookResponse(child, variables, command);
 
       return this.processFilterEquality({ payload: undefined, webhook: res }, child);
     }
 
-    return this.processFilterEquality(variables, child);
+    if (child.on === 'payload' || child.on === 'subscriber') {
+      return this.processFilterEquality(variables, child);
+    }
+
+    return false;
```

Comment:
Changed the code here a little, in the upcoming PR we will introduce the logic here for the "is online" filters.